keras/src/backend/mlx/core.py

```diff
@@ -70,9 +70,11 @@ def convert_to_tensor(x, dtype=None, sparse=None):
             return x.value.astype(mlx_dtype)
         return x.value
 
-    if isinstance(x, np.ndarray):
```

Comment:
If we remove this case, mlx crashes when running some test cases on my local machine. I logged a bug on mlx repo (see mlx backend issue).

It did not crash on CI tests on this pr because it is not being run on macOS ? Am really confused as I thought mlx runs only on macOS.

https://github.com/keras-team/keras/actions/runs/8907542811/job/24461598952#step:1:2
```
2024-05-01T09:07:10.5470064Z Current runner version: '2.316.0'
2024-05-01T09:07:10.5494922Z ##[group]Operating System
2024-05-01T09:07:10.5495654Z Ubuntu
2024-05-01T09:07:10.5496034Z 22.04.4
2024-05-01T09:07:10.5496669Z LTS
2024-05-01T09:07:10.5497089Z ##[endgroup]
2024-05-01T09:07:10.5497521Z ##[group]Runner Image
2024-05-01T09:07:10.5498174Z Image: ubuntu-22.04
2024-05-01T09:07:10.5498671Z Version: 20240422.1.0
2024-05-01T09:07:10.5499877Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240422.1/images/ubuntu/Ubuntu2204-Readme.md
2024-05-01T09:07:10.5501763Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240422.1
```